cluster-stamp.bicep

```
@@ -0,0 +1,2221 @@
+    workspaceId: law.id
+    metrics: [
+      {
+        timeGrain: 'PT1M'
+        category: 'AllMetrics'
+        enabled: true
+      }
+    ]
+    logs: [
+      {
+        category: 'ContainerRegistryRepositoryEvents'
+        enabled: true
+      }
+      {
+        category: 'ContainerRegistryLoginEvents'
+        enabled: true
+      }
+    ]
+  }
+}
+
+@description('The network interface in the spoke vnet that enables connecting privately the aks regulated cluster with cr.')
+resource peCr 'Microsoft.Network/privateEndpoints@2022-05-01' = {
+  name: 'pe-${cr.name}'
+  location: location
+  properties: {
+    subnet: {
+      id: targetVirtualNetwork::snetPrivatelinkendpoints.id
+    }
+    privateLinkServiceConnections: [
+      {
+        name: 'to-${targetVirtualNetwork.name}'
+        properties: {
+          privateLinkServiceId: cr.id
+          groupIds: [
+            'registry'
+          ]
+        }
+      }
+    ]
+  }
+  dependsOn: [
+    cr::grl
+  ]
+
+  resource pdzg 'privateDnsZoneGroups' = {
+    name: 'for-${cr.name}'
+    properties: {
+      privateDnsZoneConfigs: [
+        {
+          name: 'privatelink-azurecr-io'
+          properties: {
+            privateDnsZoneId: pdzCr.id
+          }
+        }
+      ]
+    }
+  }
+}
+
+@description('The scheduled query that returns images being imported from repositories different than quarantine/')
+resource sqrNonQuarantineImportedImgesToCr 'Microsoft.Insights/scheduledQueryRules@2022-06-15' = {
+  name: 'Image Imported into ACR from ${cr.name} source other than approved Quarantine'
+  location: location
+  properties: {
+    description: 'The only images we want in live/ are those that came from this ACR instance, but from the quarantine/ repository.'
+    actions: {
+      actionGroups: []
+    }
+    criteria: {
+      allOf: [
+        {
+          operator: 'GreaterThan'
+          query: 'ContainerRegistryRepositoryEvents\r\n| where OperationName == "importImage" and Repository startswith "live/" and MediaType !startswith strcat(_ResourceId, "/quarantine")'
+          threshold: 0
+          timeAggregation: 'Count'
+          dimensions: []
+          failingPeriods: {
+            minFailingPeriodsToAlert: 1
+            numberOfEvaluationPeriods: 1
+          }
+          resourceIdColumn: ''
+        }
+      ]
+    }
+    enabled: true
+    evaluationFrequency: 'PT10M'
+    scopes: [
+      cr.id
+    ]
+    severity: 3
+    windowSize: 'PT10M'
+    muteActionsDuration: null
+    overrideQueryTimeRange: null
+  }
+  dependsOn: [
+    cr_diagnosticSettings
+  ]
+}
+
+resource paAksLinuxRestrictive 'Microsoft.Authorization/policyAssignments@2021-06-01' = {
+  name: guid(psdAKSLinuxRestrictiveId, resourceGroup().name, clusterName)
+  properties: {
+    displayName: trim(take('[${clusterName}] ${reference(psdAKSLinuxRestrictiveId, '2020-09-01').displayName}', 125))
+    scope: subscriptionResourceId('Microsoft.Resources/resourceGroups', resourceGroup().name)
```

All of these policy assignments, I think should just be 

```suggestion
    scope: resourceGroup()
```

for scope.

---

This is for all of them, not just this one.

---

you got it @ckittel 

done | addressed from c8782f5faae65ac6e1f80143e44a987223909159

---

not the exact issue that has been raised here @ckittel , but interestingly enough I faced issues assigning a scope to these policy assignments. 

removed them from ca1e1848c2b09042140742a680de82efac15348f and leaving a comment in here in case it spark some ideas.

---

based on our offline chat/checks around the policies scope, I'm resolving this one as Im the one who re-open this.